### PR TITLE
Change the term `function` to `method`

### LINKED ISSUE #5913: Wrong terminology usage in learn pages
Double check and correct the wrong terminology in the GraphQL learn page: https://ballerina.io/learn/write-a-graphql-api-with-ballerina/

The page uses the terms "resource function" and "remote function" when it actually should be "resource method" and "remote method".

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
Change the term `function` to `method`.
> Fixes #5913 

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: components/home-page/bal-action/action-bbe/graphql-api.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ service / on new graphql:Listener(9000) {
         self.forex = check new (apiEndpoint);
     }
 
-    resource function get album(string id, Currency currency = USD) returns Album|error {
+    resource method get album(string id, Currency currency = USD) returns Album|error {
```
Comment (shafreenAnfar) [inline components/home-page/bal-action/action-bbe/graphql-api.md:39]: We shouldn't change the code.

File: public/spec/lang/2019R2/index.html [hunk 2]
```diff
@@ -5837,7 +5837,7 @@ <h3>Annotations</h3>
    <td>type of function</td>
   </tr>
   <tr>
-   <td>resource function</td>
+   <td>resource method</td>
```
Comment (MaryamZi) [inline public/spec/lang/2019R2/index.html:5840]: These shouldn't be changed either.

File: swan-lake/by-example/graphql-file-upload/content.jsx [hunk 3]
```diff
@@ -42,10 +42,10 @@ service /fileUpload on new graphql:Listener(4000) {
         return "Successfully Uploaded";
     }
 
-    // Remote functions in GraphQL services can use the \`graphql:Upload[]\` as an input parameter
-    // type. Therefore, remote functions can accept an array of \`graphql:Upload\` values. This can
+    // remote methods in GraphQL services can use the \`graphql:Upload[]\` as an input parameter
```
Comment (MaryamZi) [inline swan-lake/by-example/graphql-file-upload/content.jsx:45]:
```suggestion
    // Remote methods in GraphQL services can use the \`graphql:Upload[]\` as an input parameter
```

File: swan-lake/by-example/graphql-mutations/content.jsx [hunk 4]
```diff
@@ -24,21 +24,21 @@ service /graphql on new graphql:Listener(4000) {
...
     // name, the \`person\` object will be returned.
-    remote function updateName(string name) returns Person {
+    remote method updateName(string name) returns Person {
         self.person.setName(name);
         return self.person;
     }
 
-    // Remote function to update the age.
-    remote function updateAge(int age) returns Person {
+    // remote method to update the age.
```
Comment (MaryamZi) [inline swan-lake/by-example/graphql-mutations/content.jsx:40]:
```suggestion
    // Remote method to update the age.
```

File: components/home-page/bal-action/action-bbe/kafka-consumer-producer.md [hunk 5]
```diff
@@ -32,7 +32,7 @@ service on new kafka:Listener(kafkaEndpoint, consumerConfigs) {
         self.orderProducer = check new (kafkaEndpoint);
     }
 
-    remote function onConsumerRecord(Order[] orders) returns error? {
+    remote method onConsumerRecord(Order[] orders) returns error? {
```
Comment (pubudu91) [inline components/home-page/bal-action/action-bbe/kafka-consumer-producer.md:35]: Should not be changed in the code snippets

File: components/home-page/bal-action/action-bbe/restful-api.md [hunk 6]
```diff
@@ -17,11 +17,11 @@ table<Album> key(id) albums = table [
     ];
 
 service / on new http:Listener(port) {
-    resource function get albums() returns Album[] {
+    resource method get albums() returns Album[] {
```
Comment (pubudu91) [inline components/home-page/bal-action/action-bbe/restful-api.md:20]: Should not be changed in the code snippets

File: components/home-page/bal-action/action-bbe/working-with-databases.md [hunk 7]
```diff
@@ -24,14 +24,14 @@ service / on new http:Listener(8080) {
         self.db = check new (host, user, password, database, port);
     }
 
-    resource function get albums() returns Album[]|error? {
+    resource method get albums() returns Album[]|error? {
```
Comment (pubudu91) [inline components/home-page/bal-action/action-bbe/working-with-databases.md:27]: Should not be changed in the code snippets

File: public/spec/auth/spec.md [hunk 8]
```diff
@@ -250,7 +250,7 @@ public client class ListenerLdapUserStoreBasicAuthHandler {
         self.provider = new (config);
     }
     
-    remote function authenticate(Request|Headers|string data) returns auth:UserDetails|Unauthorized {
+    remote method authenticate(Request|Headers|string data) returns auth:UserDetails|Unauthorized {
```
Comment (pubudu91) [inline public/spec/auth/spec.md:253]: All changes in this file seem to be changes to code snippets. Should remain as `function` in code snippets.

File: public/spec/email/spec.md [hunk 9]
```diff
@@ -183,16 +183,16 @@ Both POP3 and IMAP services can be defined as follows.
 ```ballerina
 email:Service emailObserver = service object {
 
-   remote function onMessage(email:Message emailMessage) {
+   remote method onMessage(email:Message emailMessage) {
```
Comment (pubudu91) [inline public/spec/email/spec.md:186]: All changes in this file seem to be changes to code snippets. Should remain as `function` in code snippets.

File: public/spec/ftp/spec.md [hunk 10]
```diff
@@ -401,12 +401,12 @@ service ftp:Service on ftpListener {
...
 // Create a service object
 ftp:Service ftpListener = service object {
-    remote function onFileChange(ftp:WatchEvent & readonly event, ftp:Caller caller) {
+    remote method onFileChange(ftp:WatchEvent & readonly event, ftp:Caller caller) {
         // process event
     }
 };
 ```
-The remote function `onFileChange()` is invoked when the listener notices a file change in the FTP server. This function supports
+The remote method `onFileChange()` is invoked when the listener notices a file change in the FTP server. This function supports
```
Comment (pubudu91) [inline public/spec/ftp/spec.md:409]: Only this change is relevant in this file.

File: public/spec/graphql/spec.md [hunk 11]
```diff
@@ -821,11 +821,11 @@ GraphQL `Union` type represent an object that could be one of a list of GraphQL
 > **Note:** Only `distinct` service types are supported as members of a union type in the Ballerina GraphQL package. If one or more members in a union type do not follow this rule, a compilation error will be thrown.
 
 ###### Example: Union Types
-In the following example, two `distinct` service types are defined first, `Teacher` and `Student`. Then a `Union` type is defined using Ballerina syntax for defining union types. The resource function in the GraphQL service is returning the union type.
+In the following example, two `distinct` service types are defined first, `Teacher` and `Student`. Then a `Union` type is defined using Ballerina syntax for defining union types. The resource method in the GraphQL service is returning the union type.
```
Comment (pubudu91) [inline public/spec/graphql/spec.md:824]: Only changes to text such as this is relevant.

**Review by pubudu91 — COMMENTED** [review]
Looks like all the code snippets have been changed too. These should not be changed. The function -> method change is only applicable in text contexts.

**Comment by praneesha** [comment]
Closing as this has code updates as well.

File: components/home-page/bal-action/action-bbe/grpc-api.md [hunk 12]
```diff
@@ -23,12 +23,12 @@ service "Albums" on new grpc:Listener(port) {
...
     }
 
-    remote function addAlbum(Album album) returns Album|error {
+    remote method addAlbum(Album album) returns Album|error {
         albums.push(album);
         return album;
     }
 
-    remote function listAlbums() returns stream<Album, error?>|error {
+    remote method listAlbums() returns stream<Album, error?>|error {
```
[the remark this answers is not in the document]
Comment (praneesha) [reply components/home-page/bal-action/action-bbe/grpc-api.md:31]: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/6013.

File: public/spec/lang/2019R2/index.html (hunk 2 above)
Comment (praneesha) [reply public/spec/lang/2019R2/index.html:5840]: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/6013.

File: swan-lake/by-example/graphql-file-upload/content.jsx (hunk 3 above)
Comment (praneesha) [reply swan-lake/by-example/graphql-file-upload/content.jsx:45]: Need to change in the BBE source in the dist repo during the BBE hackathon.

File: swan-lake/by-example/graphql-mutations/content.jsx (hunk 4 above)
Comment (praneesha) [reply swan-lake/by-example/graphql-mutations/content.jsx:40]: Need to change in the BBE source in the dist repo during the BBE hackathon.

File: components/home-page/bal-action/action-bbe/kafka-consumer-producer.md (hunk 5 above)
Comment (praneesha) [reply components/home-page/bal-action/action-bbe/kafka-consumer-producer.md:35]: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/6013.

File: components/home-page/bal-action/action-bbe/restful-api.md (hunk 6 above)
Comment (praneesha) [reply components/home-page/bal-action/action-bbe/restful-api.md:20]: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/6013.

File: components/home-page/bal-action/action-bbe/working-with-databases.md (hunk 7 above)
Comment (praneesha) [reply components/home-page/bal-action/action-bbe/working-with-databases.md:27]: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/6013.

File: public/spec/auth/spec.md (hunk 8 above)
Comment (praneesha) [reply public/spec/auth/spec.md:253]: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/6013.

File: public/spec/email/spec.md (hunk 9 above)
Comment (praneesha) [reply public/spec/email/spec.md:186]: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/6013.

File: public/spec/ftp/spec.md (hunk 10 above)
Comment (praneesha) [reply public/spec/ftp/spec.md:409]: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/6013.

File: public/spec/graphql/spec.md (hunk 11 above)
Comment (praneesha) [reply public/spec/graphql/spec.md:824]: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/6013.

**Pull request closed without merging**